pkg/scheduler/actions/enqueue/enqueue.go

```diff
@@ -33,14 +47,21 @@ func New() *enqueueAction {
 }
 
 func (enqueue *enqueueAction) Name() string {
-	return "enqueue"
+	return EnqueueActionName
 }
 
 func (enqueue *enqueueAction) Initialize() {}
 
 func (enqueue *enqueueAction) Execute(ssn *framework.Session) {
 	glog.V(3).Infof("Enter Enqueue ...")
 	defer glog.V(3).Infof("Leaving Enqueue ...")
+	multiplier := IdleResMultiplierValue
+	if ssn.SchedStConf.Version == framework.SchedulerConfigVersion2 {
+		ret, err := getEnqueueActMultiplier(ssn.SchedStConf.V2Conf.Actions)
```

Comment:
This is not efficient.

Reply:
yeah, I do agree. but currently if we see each time during open session we are Initializing  the plugins and using the same as shown below.

I felt plugin s /action s config & initialzation has to be moved out of each seesion. I hope these things we should handle in a separaret PR. have just followed the present plug in way. what do you think?


,,,
// OpenSession start the session
func OpenSession(cache cache.Cache, schedStConf conf.SchedulerConf) *Session {
	ssn := openSession(cache, schedStConf)

	tiers := []conf.Tier{}

	if schedStConf.Version == SchedulerConfigVersion1 {
		tiers = schedStConf.V1Conf.Tiers
	} else if schedStConf.Version == SchedulerConfigVersion2 {
		tiers = schedStConf.V2Conf.Tiers
	}

	for _, tier := range tiers {
		for _, plugin := range tier.Plugins {
			if pb, found := GetPluginBuilder(plugin.Name); !found {
				glog.Errorf("Failed to get plugin %s.", plugin.Name)
			} else {
				plugin := pb(plugin.Arguments)
				ssn.plugins[plugin.Name()] = plugin
			}
		}
	}
'''

Reply:
Agreed, need a following optimization pr